libphobos/libdruntime/core/atomic.d

```diff
@@ -1415,6 +1420,11 @@ else version( GNU )
             res = __atomic_compare_exchange_8(cast(void*) here, cast(void*) &ifThis, writeThis,
                                               false, MemoryOrder.seq, MemoryOrder.seq);
         }
+        else static if( T.sizeof == long.sizeof * 2)
+        {
+            res = __atomic_compare_exchange(T.sizeof, cast(void*)here, cast(void*)&ifThis, cast(void*)&writeThis,
+                                            MemoryOrder.seq, MemoryOrder.seq);
```

Comment:
I don't see having `libatomic` as a dependency being a problem if it means that every platform benefits.  At the same time, if a best effort is made to avoid the  falling back if possible, then that is reasonable too.

That could be done either by using inline assembly, or allowing to specially hook these atomic builtins to map indirectly to `_16`.  Even though the `_16` function is not directly callable, as per reasons stated above.

Of course, getting `cent`/`ucent` pass semantic analysis would open up a lot of doors outside of just atomics. :-)